src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -0,0 +1,28 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import mate.academy.spring.service.dto.mapping.DtoResponseMapper;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserService userService;
+    private final DtoResponseMapper<UserResponseDto, User> mapper;
+
+    public UserController(UserService userService,
+                          DtoResponseMapper<UserResponseDto, User> mapper) {
+        this.userService = userService;
+        this.mapper = mapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getByEmail(@RequestParam String email) {
+        return mapper.toDto(userService.findByEmail(email).orElse(new User()));
```

Comment:
```suggestion
        return mapper.toDto(userService.findByEmail(email).get());
```

why do we need to return new User here?

Reply:
просто подумал, что если юзер не будет найден, проще пустышку возвращать, но сейчас мы ошибки прошли, так что да, незачем.